packages/squiggle-lang/src/rescript/Distributions/SampleSetDist/SampleSetDist_ToPointSet.res

```diff
@@ -70,64 +55,53 @@ let toPointSetDist = (
+    // All the same value: treat as discrete
+    Js.Array2.push(discretePart, (E.A.unsafe_get(continuousPart, 0), contLength))->ignore
+    None
+  } else {
+    // The only reason we compute _suggestedXWidth if
+    // samplingInputs.kernelWidth is given is to log it. Unnecessary?
+    let _suggestedXWidth = SampleSetDist_Bandwidth.nrd0(continuousPart)
+    let width = samplingInputs.kernelWidth->E.O.default(_suggestedXWidth)
+    let (usedWidth, pdf) =
+      continuousPart->Internals.KDE.normalSampling(
+        samplingInputs.outputXYPoints,
+        width,
+        pointWeight,
+      )
+    let samplingStats: Internals.Types.samplingStats = {
+      sampleCount: samplingInputs.sampleCount,
+      outputXYPoints: samplingInputs.outputXYPoints,
+      bandwidthXSuggested: _suggestedXWidth,
+      bandwidthXImplemented: usedWidth,
+    }
+
+    Some((pdf->Continuous.make, samplingStats))
+  }
+
   let discrete: PointSetTypes.discreteShape =
     discretePart
-    ->E.FloatFloatMap.fmap(r => r /. length, _)
-    ->E.FloatFloatMap.toArray
+    ->E.A.fmap(((x, count)) => (x, count->Belt.Int.toFloat *. pointWeight))
     ->XYShape.T.fromZippedArray
     ->Discrete.make
 
-  let pdf =
-    continuousPart->E.A.length > 5
-      ? {
-          let _suggestedXWidth = SampleSetDist_Bandwidth.nrd0(continuousPart)
-          // todo: This does some recalculating from the last step.
-          let _suggestedUnitWidth = Internals.T.suggestedUnitWidth(
-            continuousPart,
-            samplingInputs.outputXYPoints,
-          )
-          let usedWidth = samplingInputs.kernelWidth->E.O.default(_suggestedXWidth)
-          let usedUnitWidth = Internals.T.xWidthToUnitWidth(
-            samples,
-            samplingInputs.outputXYPoints,
-            usedWidth,
-          )
-          let samplingStats: Internals.Types.samplingStats = {
-            sampleCount: samplingInputs.sampleCount,
-            outputXYPoints: samplingInputs.outputXYPoints,
-            bandwidthXSuggested: _suggestedXWidth,
-            bandwidthUnitSuggested: _suggestedUnitWidth,
-            bandwidthXImplemented: usedWidth,
-            bandwidthUnitImplemented: usedUnitWidth,
-          }
-          continuousPart
-          ->Internals.T.kde(
-            ~samples=_,
-            ~outputXYPoints=samplingInputs.outputXYPoints,
-            Internals.T.formatUnitWidth(usedUnitWidth),
-          )
-          ->Continuous.make
-          ->(r => Some((r, samplingStats)))
-        }
-      : None
-
   let pointSetDist = MixedShapeBuilder.buildSimple(
     ~continuous=pdf->E.O.fmap(fst),
     ~discrete=Some(discrete),
   )
 
-  /*
-   I'm surprised that this doesn't come out normalized. My guess is that the KDE library
-  we're using is standardizing on something else. If we ever change that library, we should
-  check to see if we still need to do this.
- */
-
-  let normalizedPointSet = pointSetDist->E.O.fmap(PointSetDist.T.normalize)
```

Comment:
Are you sure that the output of the new alg is normalized? My guess is that you have to normalize in respect to an interpolation algorithm - now we're using linear interpolation, so we'd need to do at least a bit of adjustment accordingly.

Reply:
Just flagging that if/when we change our interpolation strategy, then I assume this will stop being normalized.